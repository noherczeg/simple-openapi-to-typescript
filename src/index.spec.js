const fs = require('fs');
const path = require('path');
const generator = require('./index');

const odataOpenApiPath = './examples/odata-openapi/northwind';
const openApiPetStorePath = './examples/OpenAPI-Specification/petstore';
const openApiPetStoreWithOptionsPath = './examples/OpenAPI-Specification/petstore-with-options';
const openApiPetStoreExpPath = './examples/OpenAPI-Specification/petstore-expanded';
const openApiLinkPath = './examples/OpenAPI-Specification/link';
const openApiUsptoPath = './examples/OpenAPI-Specification/uspto';
const schemaOutputFor = (root, fileName) => fs.readFileSync(`${root}/components/schemas/${fileName}`).toString();
const pathOutputFor = (root, fileName) => fs.readFileSync(`${root}/paths/${fileName}`).toString();

describe('openapi-web-gen', () => {
  describe('flags', () => {
    beforeAll(() => {
      generator(path.resolve(__dirname, '__fixtures__/OpenAPI-Specification/examples/v3.0/petstore.yaml'), {
        target: openApiPetStoreWithOptionsPath,
        prettierOpts: path.resolve(__dirname, '__fixtures__/OpenAPI-Specification/prettier-custom.json'),
      });
    });

    it('common/Common.ts', () => {
      expect(fs.readFileSync(`${openApiPetStoreWithOptionsPath}/common/Common.ts`).toString()).toMatchSnapshot();
    });
  });

  describe('odata-openapi/Northwind.openapi3.json should generate', () => {
    beforeAll(() => {
      generator(path.resolve(__dirname, '__fixtures__/odata-openapi/examples/Northwind.openapi3.json'), { target: odataOpenApiPath });
    });

    it('common/Common.ts', () => {
      expect(fs.readFileSync(`${odataOpenApiPath}/common/Common.ts`).toString()).toMatchSnapshot();
    });

    it('components/responses/Error.ts', () => {
      expect(fs.readFileSync(`${odataOpenApiPath}/components/responses/Error.ts`).toString()).toMatchSnapshot();
    });

    describe('components/schemas/', () => {
      it('NorthwindModelAlphabeticalListOfProduct.ts', () => {
        expect(schemaOutputFor(odataOpenApiPath, 'NorthwindModelAlphabeticalListOfProduct.ts')).toMatchSnapshot();
      });

      it('NorthwindModelCategory.ts', () => {
        expect(schemaOutputFor(odataOpenApiPath, 'NorthwindModelCategory.ts')).toMatchSnapshot();
      });

      it('NorthwindModelCurrentProductListCreate.ts', () => {
        expect(schemaOutputFor(odataOpenApiPath, 'NorthwindModelCurrentProductListCreate.ts')).toMatchSnapshot();
      });

      it('NorthwindModelCustomer.ts', () => {
        expect(schemaOutputFor(odataOpenApiPath, 'NorthwindModelCustomer.ts')).toMatchSnapshot();
      });

      it('NorthwindModelEmployee.ts', () => {
        expect(schemaOutputFor(odataOpenApiPath, 'NorthwindModelEmployee.ts')).toMatchSnapshot();
      });

      it('NorthwindModelOrderDetail.ts', () => {
        expect(schemaOutputFor(odataOpenApiPath, 'NorthwindModelOrderDetail.ts')).toMatchSnapshot();
      });
    });

    describe('paths/', () => {
      it('GetEntitiesFromSalesTotalsByAmounts.ts', () => {
        expect(pathOutputFor(odataOpenApiPath, 'GetEntitiesFromSalesTotalsByAmounts.ts'))
          .toMatchSnapshot();
      });

      it('GetEntitiesFromRelatedProducts.ts', () => {
        expect(pathOutputFor(odataOpenApiPath, 'GetEntitiesFromRelatedProducts.ts'))
          .toMatchSnapshot();
      });

      it('UpdateEntityInAlphabeticalListOfProducts.ts', () => {
        expect(pathOutputFor(odataOpenApiPath, 'UpdateEntityInAlphabeticalListOfProducts.ts'))
          .toMatchSnapshot();
      });
    });
  });

  describe('OpenAPI-Specification/examples/v3.0/petstore.yaml should generate', () => {
    beforeAll(() => {
      generator(path.resolve(__dirname, '__fixtures__/OpenAPI-Specification/examples/v3.0/petstore.yaml'), { target: openApiPetStorePath });
    });

    it('common/Common.ts', () => {
      expect(fs.readFileSync(`${openApiPetStorePath}/common/Common.ts`).toString()).toMatchSnapshot();
    });

    describe('components/schemas/', () => {
      it('Error.ts', () => {
        expect(schemaOutputFor(openApiPetStorePath, 'Error.ts')).toMatchSnapshot();
      });

      it('Pet.ts', () => {
        expect(schemaOutputFor(openApiPetStorePath, 'Pet.ts')).toMatchSnapshot();
      });

      it('Pets.ts', () => {
        expect(schemaOutputFor(openApiPetStorePath, 'Pets.ts')).toMatchSnapshot();
      });
    });
  });

  describe('OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml should generate', () => {
    beforeAll(() => {
      generator(path.resolve(__dirname, '__fixtures__/OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml'), { target: openApiPetStoreExpPath });
    });

    it('common/Common.ts', () => {
      expect(fs.readFileSync(`${openApiPetStoreExpPath}/common/Common.ts`).toString()).toMatchSnapshot();
    });

    describe('components/schemas/', () => {
      it('Error.ts', () => {
        expect(schemaOutputFor(openApiPetStoreExpPath, 'Error.ts')).toMatchSnapshot();
      });

      it('NewPet.ts', () => {
        expect(schemaOutputFor(openApiPetStoreExpPath, 'NewPet.ts')).toMatchSnapshot();
      });

      it('Pet.ts', () => {
        expect(schemaOutputFor(openApiPetStoreExpPath, 'Pet.ts')).toMatchSnapshot();
      });
    });

    describe('paths/', () => {
      it('AddPet.ts', () => {
        expect(pathOutputFor(openApiPetStoreExpPath, 'AddPet.ts')).toMatchSnapshot();
      });
    });
  });

  describe('OpenAPI-Specification/examples/v3.0/link-example.yaml should generate', () => {
    beforeAll(() => {
      generator(path.resolve(__dirname, '__fixtures__/OpenAPI-Specification/examples/v3.0/link-example.yaml'), { target: openApiLinkPath });
    });

    describe('components/schemas/', () => {
      it('User.ts', () => {
        expect(schemaOutputFor(openApiLinkPath, 'User.ts')).toMatchSnapshot();
      });

      it('Repository.ts', () => {
        expect(schemaOutputFor(openApiLinkPath, 'User.ts')).toMatchSnapshot();
      });

      it('Pullrequest.ts', () => {
        expect(schemaOutputFor(openApiLinkPath, 'User.ts')).toMatchSnapshot();
      });
    });
  });

  describe('OpenAPI-Specification/examples/v3.0/uspto.yaml should generate', () => {
    beforeAll(() => {
      generator(path.resolve(__dirname, '__fixtures__/OpenAPI-Specification/examples/v3.0/uspto.yaml'), { target: openApiUsptoPath });
    });

    it('common/Common.ts', () => {
      expect(fs.readFileSync(`${openApiUsptoPath}/common/Common.ts`).toString()).toMatchSnapshot();
    });

    describe('components/schemas/', () => {
      it('DataSetList.ts', () => {
        expect(schemaOutputFor(openApiUsptoPath, 'DataSetList.ts')).toMatchSnapshot();
      });
    });
  });
});
