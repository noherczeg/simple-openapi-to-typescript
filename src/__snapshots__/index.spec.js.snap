// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/link-example.yaml should generate components/schemas/ Pullrequest.ts 1`] = `
"/* Generated source, do not modify! */

export interface User {
  username?: string;

  uuid?: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/link-example.yaml should generate components/schemas/ Repository.ts 1`] = `
"/* Generated source, do not modify! */

export interface User {
  username?: string;

  uuid?: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/link-example.yaml should generate components/schemas/ User.ts 1`] = `
"/* Generated source, do not modify! */

export interface User {
  username?: string;

  uuid?: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore.yaml should generate common/Common.ts 1`] = `
"/* Generated source, do not modify! */

export interface ServerVariableObject {
  enum?: string[];
  default: string;
  description?: string;
}

export interface ServerObject {
  url: string;
  description?: string;
  variables?: Map<string, ServerVariableObject>;
}

export type HostNameMapper = (servers: ServerObject[]) => string;

export const servers: ServerObject[] = [
  {
    url: \\"http://petstore.swagger.io/v1\\"
  }
];

export function createHostName(mapper: HostNameMapper): string {
  return mapper(servers);
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore.yaml should generate components/schemas/ Error.ts 1`] = `
"/* Generated source, do not modify! */

export interface Error {
  /**
   * format: int32
   */
  code: number;

  message: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore.yaml should generate components/schemas/ Pet.ts 1`] = `
"/* Generated source, do not modify! */

export interface Pet {
  /**
   * format: int64
   */
  id: number;

  name: string;

  tag?: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore.yaml should generate components/schemas/ Pets.ts 1`] = `
"/* Generated source, do not modify! */

import { Pet } from \\"./Pet\\";

export interface Pets extends Array<Pet> {}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml should generate common/Common.ts 1`] = `
"/* Generated source, do not modify! */

export interface ServerVariableObject {
  enum?: string[];
  default: string;
  description?: string;
}

export interface ServerObject {
  url: string;
  description?: string;
  variables?: Map<string, ServerVariableObject>;
}

export type HostNameMapper = (servers: ServerObject[]) => string;

export const servers: ServerObject[] = [
  {
    url: \\"http://petstore.swagger.io/api\\"
  }
];

export function createHostName(mapper: HostNameMapper): string {
  return mapper(servers);
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml should generate components/schemas/ Error.ts 1`] = `
"/* Generated source, do not modify! */

export interface Error {
  /**
   * format: int32
   */
  code: number;

  message: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml should generate components/schemas/ NewPet.ts 1`] = `
"/* Generated source, do not modify! */

export interface NewPet {
  name: string;

  tag?: string;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml should generate components/schemas/ Pet.ts 1`] = `
"/* Generated source, do not modify! */

import { NewPet } from \\"./NewPet\\";

export interface Pet extends NewPet {
  /**
   * format: int64
   */
  id: number;
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/petstore-expanded.yaml should generate paths/ PetsPost.ts 1`] = `
"/* Generated source, do not modify! */

import { HttpMethods } from \\"../constants/HttpMethods\\";
import { MediaTypes } from \\"../constants/MediaTypes\\";

export function createPath(): string {
  return \`/pets\`;
}

export const method: HttpMethods = HttpMethods.POST;

export const requestContentType: MediaTypes = MediaTypes.APPLICATION_JSON;
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/uspto.yaml should generate common/Common.ts 1`] = `
"/* Generated source, do not modify! */

export interface ServerVariableObject {
  enum?: string[];
  default: string;
  description?: string;
}

export interface ServerObject {
  url: string;
  description?: string;
  variables?: Map<string, ServerVariableObject>;
}

export type HostNameMapper = (servers: ServerObject[]) => string;

export const servers: ServerObject[] = [
  {
    url: \\"\${scheme}://developer.uspto.gov/ds-api\\",

    variables: new Map<string, ServerVariableObject>([
      [
        \\"scheme\\",
        {
          default: \\"https\\",
          description: \\"The Data Set API is accessible via https and http\\",
          enum: [\\"https\\", \\"http\\"]
        }
      ]
    ])
  }
];

export function createHostName(mapper: HostNameMapper): string {
  return mapper(servers);
}
"
`;

exports[`openapi-web-gen OpenAPI-Specification/examples/v3.0/uspto.yaml should generate components/schemas/ DataSetList.ts 1`] = `
"/* Generated source, do not modify! */

export interface DataSetList {
  total?: number;

  apis?: Array<any>;
}
"
`;

exports[`openapi-web-gen flags common/Common.ts 1`] = `
"/* Generated source, do not modify! */

export interface ServerVariableObject {
    enum?: string[];
    default: string;
    description?: string;
}

export interface ServerObject {
    url: string;
    description?: string;
    variables?: Map<string, ServerVariableObject>;
}

export type HostNameMapper = (servers: ServerObject[]) => string;

export const servers: ServerObject[] = [
    {
        url: 'http://petstore.swagger.io/v1'
    }
];

export function createHostName(mapper: HostNameMapper): string {
    return mapper(servers);
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate common/Common.ts 1`] = `
"/* Generated source, do not modify! */

export interface ServerVariableObject {
  enum?: string[];
  default: string;
  description?: string;
}

export interface ServerObject {
  url: string;
  description?: string;
  variables?: Map<string, ServerVariableObject>;
}

export type HostNameMapper = (servers: ServerObject[]) => string;

export const servers: ServerObject[] = [
  {
    url: \\"https://services.odata.org/V4/Northwind/Northwind.svc\\"
  }
];

export function createHostName(mapper: HostNameMapper): string {
  return mapper(servers);
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/responses/Error.ts 1`] = `
"/* Generated source, do not modify! */

export interface Error {
  error: any;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ Error.ts 1`] = `
"/* Generated source, do not modify! */

export interface Error {
  error: any;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ NorthwindModelAlphabeticalListOfProduct.ts 1`] = `
"/* Generated source, do not modify! */

export interface NorthwindModelAlphabeticalListOfProduct {
  /**
   * format: int32
   */
  ProductID?: number;

  /**
   * maxLength: 40
   */
  ProductName?: string;

  /**
   * format: int32
   */
  SupplierID?: number;

  /**
   * format: int32
   */
  CategoryID?: number;

  /**
   * maxLength: 20
   */
  QuantityPerUnit?: string;

  /**
   * format: decimal
   * multipleOf: 0.0001
   */
  UnitPrice?: number | string;

  /**
   * format: int16
   */
  UnitsInStock?: number;

  /**
   * format: int16
   */
  UnitsOnOrder?: number;

  /**
   * format: int16
   */
  ReorderLevel?: number;

  Discontinued?: boolean;

  /**
   * maxLength: 15
   */
  CategoryName?: string;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ NorthwindModelCategory.ts 1`] = `
"/* Generated source, do not modify! */

import { NorthwindModelProduct } from \\"./NorthwindModelProduct\\";

export interface NorthwindModelCategory {
  /**
   * format: int32
   */
  CategoryID?: number;

  /**
   * maxLength: 15
   */
  CategoryName?: string;

  Description?: string;

  /**
   * format: base64url
   */
  Picture?: string;

  Products?: Array<NorthwindModelProduct>;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ NorthwindModelCurrentProductListCreate.ts 1`] = `
"/* Generated source, do not modify! */

export interface NorthwindModelCurrentProductListCreate {
  /**
   * format: int32
   */
  ProductID: number;

  /**
   * maxLength: 40
   */
  ProductName: string;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ NorthwindModelCustomer.ts 1`] = `
"/* Generated source, do not modify! */

import { NorthwindModelOrder } from \\"./NorthwindModelOrder\\";
import { NorthwindModelCustomerDemographic } from \\"./NorthwindModelCustomerDemographic\\";

export interface NorthwindModelCustomer {
  /**
   * maxLength: 5
   */
  CustomerID?: string;

  /**
   * maxLength: 40
   */
  CompanyName?: string;

  /**
   * maxLength: 30
   */
  ContactName?: string;

  /**
   * maxLength: 30
   */
  ContactTitle?: string;

  /**
   * maxLength: 60
   */
  Address?: string;

  /**
   * maxLength: 15
   */
  City?: string;

  /**
   * maxLength: 15
   */
  Region?: string;

  /**
   * maxLength: 10
   */
  PostalCode?: string;

  /**
   * maxLength: 15
   */
  Country?: string;

  /**
   * maxLength: 24
   */
  Phone?: string;

  /**
   * maxLength: 24
   */
  Fax?: string;

  Orders?: Array<NorthwindModelOrder>;

  CustomerDemographics?: Array<NorthwindModelCustomerDemographic>;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ NorthwindModelEmployee.ts 1`] = `
"/* Generated source, do not modify! */

import { NorthwindModelEmployee } from \\"./NorthwindModelEmployee\\";
import { NorthwindModelOrder } from \\"./NorthwindModelOrder\\";
import { NorthwindModelTerritory } from \\"./NorthwindModelTerritory\\";

export interface NorthwindModelEmployee {
  /**
   * format: int32
   */
  EmployeeID?: number;

  /**
   * maxLength: 20
   */
  LastName?: string;

  /**
   * maxLength: 10
   */
  FirstName?: string;

  /**
   * maxLength: 30
   */
  Title?: string;

  /**
   * maxLength: 25
   */
  TitleOfCourtesy?: string;

  /**
   * format: date-time
   */
  BirthDate?: string;

  /**
   * format: date-time
   */
  HireDate?: string;

  /**
   * maxLength: 60
   */
  Address?: string;

  /**
   * maxLength: 15
   */
  City?: string;

  /**
   * maxLength: 15
   */
  Region?: string;

  /**
   * maxLength: 10
   */
  PostalCode?: string;

  /**
   * maxLength: 15
   */
  Country?: string;

  /**
   * maxLength: 24
   */
  HomePhone?: string;

  /**
   * maxLength: 4
   */
  Extension?: string;

  /**
   * format: base64url
   */
  Photo?: string;

  Notes?: string;

  /**
   * format: int32
   */
  ReportsTo?: number;

  /**
   * maxLength: 255
   */
  PhotoPath?: string;

  Employees1?: Array<NorthwindModelEmployee>;

  Employee1?: NorthwindModelEmployee;

  Orders?: Array<NorthwindModelOrder>;

  Territories?: Array<NorthwindModelTerritory>;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate components/schemas/ NorthwindModelOrderDetail.ts 1`] = `
"/* Generated source, do not modify! */

import { NorthwindModelOrder } from \\"./NorthwindModelOrder\\";
import { NorthwindModelProduct } from \\"./NorthwindModelProduct\\";

export interface NorthwindModelOrderDetail {
  /**
   * format: int32
   */
  OrderID?: number;

  /**
   * format: int32
   */
  ProductID?: number;

  /**
   * format: decimal
   * multipleOf: 0.0001
   */
  UnitPrice?: number | string;

  /**
   * format: int16
   */
  Quantity?: number;

  /**
   * format: float
   */
  Discount?: number | string;

  Order?: NorthwindModelOrder;

  Product?: NorthwindModelProduct;
}
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate paths/ CategoriesCategoryIDProductsGet.ts 1`] = `
"/* Generated source, do not modify! */

import { HttpMethods } from \\"../constants/HttpMethods\\";

/**
 * description: Order items by property values, see [Sorting](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptionorderby)
 */
export enum OrderbyEnum {
  ProductID = \\"ProductID\\",
  ProductIDDesc = \\"ProductID desc\\",
  ProductName = \\"ProductName\\",
  ProductNameDesc = \\"ProductName desc\\",
  SupplierID = \\"SupplierID\\",
  SupplierIDDesc = \\"SupplierID desc\\",
  CategoryID = \\"CategoryID\\",
  CategoryIDDesc = \\"CategoryID desc\\",
  QuantityPerUnit = \\"QuantityPerUnit\\",
  QuantityPerUnitDesc = \\"QuantityPerUnit desc\\",
  UnitPrice = \\"UnitPrice\\",
  UnitPriceDesc = \\"UnitPrice desc\\",
  UnitsInStock = \\"UnitsInStock\\",
  UnitsInStockDesc = \\"UnitsInStock desc\\",
  UnitsOnOrder = \\"UnitsOnOrder\\",
  UnitsOnOrderDesc = \\"UnitsOnOrder desc\\",
  ReorderLevel = \\"ReorderLevel\\",
  ReorderLevelDesc = \\"ReorderLevel desc\\",
  Discontinued = \\"Discontinued\\",
  DiscontinuedDesc = \\"Discontinued desc\\"
}

/**
 * description: Select properties to be returned, see [Select](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptionselect)
 */
export enum SelectEnum {
  ProductID = \\"ProductID\\",
  ProductName = \\"ProductName\\",
  SupplierID = \\"SupplierID\\",
  CategoryID = \\"CategoryID\\",
  QuantityPerUnit = \\"QuantityPerUnit\\",
  UnitPrice = \\"UnitPrice\\",
  UnitsInStock = \\"UnitsInStock\\",
  UnitsOnOrder = \\"UnitsOnOrder\\",
  ReorderLevel = \\"ReorderLevel\\",
  Discontinued = \\"Discontinued\\"
}

/**
 * description: Expand related entities, see [Expand](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptionexpand)
 */
export enum ExpandEnum {
  _ALL_ = \\"*\\",
  Category = \\"Category\\",
  OrderDetails = \\"Order_Details\\",
  Supplier = \\"Supplier\\"
}

export interface PathParams {
  CategoryID: number;
}

export interface RequestSearchParams {
  $filter?: string;
  $orderby?: Set<OrderbyEnum>;
  $select?: Set<SelectEnum>;
  $expand?: Set<ExpandEnum>;
}

export function createPath(pathParams: PathParams): string {
  return \`/Categories(\${pathParams.CategoryID})/Products\`;
}

export const method: HttpMethods = HttpMethods.GET;
"
`;

exports[`openapi-web-gen odata-openapi/Northwind.openapi3.json should generate paths/ SalesTotalsByAmountsGet.ts 1`] = `
"/* Generated source, do not modify! */

import { HttpMethods } from \\"../constants/HttpMethods\\";

/**
 * description: Order items by property values, see [Sorting](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptionorderby)
 */
export enum OrderbyEnum {
  SaleAmount = \\"SaleAmount\\",
  SaleAmountDesc = \\"SaleAmount desc\\",
  OrderID = \\"OrderID\\",
  OrderIDDesc = \\"OrderID desc\\",
  CompanyName = \\"CompanyName\\",
  CompanyNameDesc = \\"CompanyName desc\\",
  ShippedDate = \\"ShippedDate\\",
  ShippedDateDesc = \\"ShippedDate desc\\"
}

/**
 * description: Select properties to be returned, see [Select](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptionselect)
 */
export enum SelectEnum {
  SaleAmount = \\"SaleAmount\\",
  OrderID = \\"OrderID\\",
  CompanyName = \\"CompanyName\\",
  ShippedDate = \\"ShippedDate\\"
}

export interface RequestSearchParams {
  $filter?: string;
  $orderby?: Set<OrderbyEnum>;
  $select?: Set<SelectEnum>;
}

export function createPath(): string {
  return \`/Sales_Totals_by_Amounts\`;
}

export const method: HttpMethods = HttpMethods.GET;
"
`;
